server/rooms.ts

```diff
@@ -719,10 +719,8 @@ export abstract class BasicRoom {
 			}
 		}
 
-		if (this.persist) {
-			this.title = newTitle;
-			this.saveSettings();
-		}
+		this.settings.title = newTitle;
+		this.saveSettings();
```

Comment:
This is probably a good change, but I'm curious how it constitutes a bug. We shouldn't be reading `this.settings.title` except on room creation, which a non-persistent room shouldn't be doing, right?

Reply:
> While there is a banner stating the room has been renamed, it's not a link -- it seems that all user connections are moved to the new room, as is the modlog. There is, however, a bug in `/renameroom`, which I opened a PR to fix (#6952): the chatroom's new name is never saved to `chatrooms.json`, and thus the rename partially reverts after restarting.

Oh, I understand now! I think this implies a worse underlying bug: that `room.persist` isn't being set correctly? This will need more investigation.

Reply:
The bug I found was that it was just setting `this.title`, which never actually gets saved.

Reply:
(and since `saveSettings()` checks `room.persist`, I found it silly to keep the conditional in there)

Reply:
Oh! I didn't even notice that until now.